### DISCUSSION TITLE
Does the use of the InMemory store for test preclude me from doing async test cases?

### DISCUSSION THREAD

**Discussion by zorn** [commanded/commanded#499] [Q&A] [answered]
If I am using the `Commanded.EventStore.Adapters.InMemory` store during tests and resetting it in `on_exit` as this [wiki page](https://github.com/commanded/commanded/wiki/In-memory-event-store#exunit-case-template) suggests, does this preclude me from using the `async: true` on the ExUnit test case? I assume the `InMemory` store is global state, right?

Also, in that wiki page, I feel like it is missing the actual call to `reset/2`. I don't seem to have the power to suggest edits to the wiki so if you see this and agree, maybe consider updating it.

```diff
 defmodule InMemoryEventStoreCase do
   use ExUnit.CaseTemplate
 
   alias Commanded.EventStore.Adapters.InMemory
 
   setup do
     {:ok, _apps} = Application. ensure_all_started(:my_app)
    
+     Commanded.EventStore.Adapters.InMemory.reset!(MyApp.App)
 
     on_exit(fn ->
       :ok = Application.stop(:my_app)
     end)
   end
 end
```

Thanks for any help or insights. Enjoying the library so far with my tinker project.

**Comment by jwilger** [comment, marked as answer]
That is correct. It's the same behavior as the DB-backed event store; it just never persists any events outside of system memory.

Personally, I'd suggest not bothering with the InMemory store. The majority of your tests should be pure functions in order to keep things simple and enable async execution. Anything where you *need* to test end-to-end behavior would be better served by using the same event store you will use in production, otherwise you're test is only proving it works in a situation that doesn't exist in production.
